### PR TITLE
[INFRA/IIGO] RE: Issue 224 - Logging in IIGO 

### PR DESCRIPTION
**Pull request by preetl**
# Summary

EDIT: Logs added for IIGO. I haven't logged everything as quite a lot of actions happen in IIGO but we can change this if necessary.

Just a test to work out how this would be possible. @lhl2617 would you mind telling me if this fits what you had described in issue #224 ? I'm not sure if it's the right way to do it but it does seem to work. Will add in more helpful logging statements if it's correct.

## Additional Information

N/A

## Test Plan

N/A

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

File: internal/server/iigo.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ func (s *SOMASServer) runIIGO() error {
 
 	nonDead := getNonDeadClientIDs(s.gameState.ClientInfos)
 	updateAliveIslands(nonDead)
-	iigoSuccessful, iigoStatus := iigointernal.RunIIGO(&s.gameState, &s.clientMap)
+	iigoSuccessful, iigoStatus := iigointernal.RunIIGO(s.iigoServer, &s.gameState, &s.clientMap)
```
Comment (lhl2617) [inline internal/server/iigo.go]: you can just pass in `s.Logf` i think

File: internal/server/iigointernal/orchestration.go [hunk 2]
```diff
@@ -9,6 +9,14 @@ import (
 	"github.com/SOMAS2020/SOMAS2020/internal/common/voting"
 )
 
+type IIGOServer struct {
+	IIGOLogging func(format string, a ...interface{})
```
Comment (lhl2617) [inline internal/server/iigointernal/orchestration.go]: logger is a nicer name :) not need to export as well

**Comment by lhl2617** [comment]
yes, this is correct. you should now add loggers to your `judiciary` etc. struct that uses the server's logger.

even better that you can print in the format
`[SERVER][JUDICIARY]: ...` for example

File: internal/server/iigo.go (hunk 1 above)
Comment (preetl) [reply internal/server/iigo.go]: oh so no need to make the iigoserver struct ? lol

Comment (preetl) [reply internal/server/iigo.go]: or do you mean s.iigoServer.logf ?

Comment (lhl2617) [reply internal/server/iigo.go]: `s.logf` is a logger function you can pass in

**Comment by lhl2617** [comment]
image

indeed

**Comment by lhl2617** [comment]
the logger type is actually same for all loggers, so perhaps we can move it into shared.

**Comment by NGovani** [comment]
I think he meant that since the three roles perform transactions such as taking from the common pool and giving resources to each other, these transactions should be logged correctly. In order to do that the role objects need access to the logger

**Comment by lhl2617** [comment]
See #255

**Comment by lhl2617** [comment]
is this active? @preetl

**Comment by preetl** [comment]
> is this active? @preetl

I'm just adding in logging statements, then I'll make it ready for review!

File: internal/server/iigointernal/judiciary.go [hunk 3]
```diff
@@ -235,6 +241,7 @@ func (j *judiciary) scoreIslandTransgressions(transgressions map[shared.ClientID
 			} else {
 				totalIslandTurnScore += roles.IIGOSanctionScore(1)
 			}
+			j.Logf("Rule: %v, broken by %v", ruleBroken, islandID)
```
Comment (Neelesh99) [inline internal/server/iigointernal/judiciary.go]: @ramon15749 : What's this inconsistency with the colons @preetl ?

**Review by Neelesh99 — CHANGES_REQUESTED** [review]
Almost there, ramon had a comment for ya

**Review by Neelesh99 — APPROVED** [review]
LGTM

**Pull request merged**